plans/network/pingpong.go

```diff
@@ -187,6 +206,96 @@ func pingpong(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 	return nil
 }
 
+// Exchanges own data network address with other instance in the test, and returns their data addres
+func exchangeListenAddrs(ctx context.Context, runenv *runtime.RunEnv, client sync.Client) (string, error) {
+
+	// Retrieve own data IP
+	tcpAddr, err := getSubnetAddr(runenv.TestSubnet)
+	if err != nil {
+		return "", err
+	}
+
+	mynode := &ListenAddrs{}
+	mine := map[string]struct{}{}
+
+	mynode.Addrs = append(mynode.Addrs, tcpAddr.String())
+	for _, l := range mynode.Addrs {
+		mine[l] = struct{}{}
+	}
```

Comment:
See the comment below, if you have a single address, can you get rid of these constructions?